File: src/features/typing/data/go/best_time_to_buy_and_sell_stock.go
package main

func maxProfit(prices []int) int {
	minPrice := prices[0]
	best := 0
	for _, price := range prices {
		if price < minPrice {
			minPrice = price
		} else if price-minPrice > best {
			best = price - minPrice
		}
	}
	return best
}

File: src/features/typing/data/java/best_time_to_buy_and_sell_stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = prices[0];
        int best = 0;
        for (int price : prices) {
            if (price < minPrice) {
                minPrice = price;
            } else {
                best = Math.max(best, price - minPrice);
            }
        }
        return best;
    }
}
